src/main/java/org/spongepowered/api/service/economy/transaction/EconomyTransactionResultType.java

```diff
@@ -24,36 +24,33 @@
  */
 package org.spongepowered.api.service.economy.transaction;
 
-import org.spongepowered.api.service.context.Context;
+import org.spongepowered.api.CatalogType;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.util.annotation.CatalogedBy;
+
+import java.util.Optional;
 
 /**
- * Represents the success or failure state of a {@link TransactionResult}.
+ * A catalog type which signifies the result of an economy transaction.
  */
-public enum ResultType {
+@CatalogedBy(EconomyTransactionResultTypes.class)
+public interface EconomyTransactionResultType extends CatalogType {
 
     /**
-     * Transaction finished with no errors.
+     * Gets whether the economy transaction was successfully completed.
+     *
+     * @return If the economy transaction was successful.
      */
-    SUCCESS,
+    boolean isSuccessful();
 
     /**
-     * The transaction attempt resulted in a difference in {@link Context}s.
+     * Gets a default message tied to the
+     * specific result type if specified.
+     *
+     * @return A default message tied to the result type
      */
-    CONTEXT_MISMATCH,
+    default Optional<Text> getMessage() {
```

Comment:
Don't bother defaulting this - developers should at least make an explicit choice here _especially_ as they likely to be adding error results, rather than success results.